crates/orderbook/src/api/get_native_prices.rs

```diff
@@ -0,0 +1,70 @@
+use crate::{api::internal_error, orderbook::Orderbook};
+use anyhow::{anyhow, Result};
+use ethcontract::H160;
+use serde::Serialize;
+use shared::api::{ApiReply, IntoWarpReply};
+use std::{convert::Infallible, sync::Arc};
+use warp::{hyper::StatusCode, reply::with_status, Filter, Rejection};
+
+#[derive(Serialize)]
+struct PriceResponse {
+    price: f64,
+}
+
+impl From<f64> for PriceResponse {
+    fn from(price: f64) -> Self {
+        Self { price }
+    }
+}
+
+fn get_native_prices_request() -> impl Filter<Extract = (H160,), Error = Rejection> + Clone {
+    warp::path!("token" / H160 / "native_prices").and(warp::get())
+}
+
+pub fn get_native_prices(
+    orderbook: Arc<Orderbook>,
```

Comment:
Instead of passing the whole `Orderbook`, should we just pass `Arc<dyn NativePriceEstimating>`?

Reply:
Fixed. Implemented in the simplest way now.